### PR TITLE
Make existing minder CLI work for VerifyProviderToken

### PR DESCRIPTION
**Pull request by evankanderson**
# Summary

The existing `minder provider enroll` CLI polls `VerifyProviderTokenFrom` looking for token
saved after the command launched.  That's not quite how GitHub Apps work, but we can look
for newly registered apps of the correct type that were created since the requested timestamp.

This allows running `minder provider enroll --provider github-app` with a `minder` binary
that's new enough not to have hard-coded the check of known providers into the client.

## Change Type

***Mark the type of change your PR introduces:***

- [x] Bug fix (resolves an issue without affecting existing features)
- [ ] Feature (adds new functionality without breaking changes)
- [ ] Breaking change (may impact existing functionalities or require documentation updates)
- [ ] Documentation (updates or additions to documentation)
- [ ] Refactoring or test improvements (no bug fixes or new functionality)

# Testing

Set up a new docker environment with a GitHub app used for both KeyCloak and Minder, with the following callback urls:

- `http://localhost:8080/api/v1/auth/callback/github-app/app`
- `http://localhost:8081/realms/stacklok/broker/github/endpoint`

**NOTE THAT ORDER MATTERS**

I also found that the GitHub App seemed to be fairly particular, and didn't want to allow `Exchange` for `http://localhost:8080/api/v1/auth/callback/github-app/app/cli` with the `http://localhost:8080/api/v1/auth/callback/github-app/app` URL.  _I_ fixed that by removing the `/cli` or `/web` suffix, but we might want to fix it a different way (e.g. by registering multiple allowed callback URLs).

# Review Checklist:

- [x] Reviewed my own code for quality and clarity.
- [x] Added comments to complex or tricky code sections.
- [ ] Updated any affected documentation.
- [ ] Included tests that validate the fix or feature.
- [x] Checked that related changes are merged.

### CODE REVIEW

**Comment by evankanderson** [comment]
This has a similar intent, though different implementation, than #2882

**Review by eleftherias — DISMISSED** [review]
Had a discussion with @evankanderson offline and decided this will be good to merge for backwards compatibility with old CLI versions, independent of https://github.com/stacklok/minder/pull/2882

File: internal/controlplane/handlers_oauth.go [hunk 1]
```diff
@@ -448,6 +449,39 @@ func (s *Server) VerifyProviderTokenFrom(ctx context.Context,
...
+	// in the meantime.
+	if in.GetContext().GetProvider() == string(db.ProviderClassGithubApp) {
+		// TODO: factor this into a method someplace better.
+
+		// We don't know which app the user is trying to register, but see if there's
+		// a recently-created one.
+		prov, err := s.store.FindProviders(ctx, db.FindProvidersParams{
+			Projects: []uuid.UUID{projectID},
+			Trait: db.NullProviderType{
+				ProviderType: db.ProviderTypeGithub,
```
Comment (JAORMX) [inline internal/controlplane/handlers_oauth.go:468]: didn't we have an explicit trait for github apps?

Comment (eleftherias) [reply internal/controlplane/handlers_oauth.go:468]: No, we just have the class

**Review by JAORMX — DISMISSED** [review]

**Review by JAORMX — APPROVED** [review]

**Pull request merged**